js/admin/retire-form.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import Cookies from "js-cookie";
+
+class RetireForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      submitSuccess: false,
+      retirementDate: null,
+      reason: null,
+      isDead: false,
+      vacantSeat: false,
+    }
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleInputChange(event) {
+    const target = event.target;
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    const name = target.name;
+
+    this.setState({
+      [name]: value
+    });
+  }
+
+  handleSubmit(event) {
+    const csrftoken = Cookies.get("csrftoken");
+    const url = "/admin/people/retire/";
+    const {retirementDate, reason, isDead, vacantSeat} = this.state;
+    const reasoning = isDead ? 'Death' : reason;
```

Comment:
Tried this out and it seems like the reason field is required even if death is checked, could we have checking it just populate the reason field?

Reply:
Yeah, removed the requirement and checked that it was correct if the death checkbox was clicked multiple times.